lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py

```diff
@@ -28,69 +28,60 @@ def get_text(self, image: Image.Image) -> str:
+                    obj_bbox.y1 / height,
+                    obj_bbox.x2 / width,
+                    obj_bbox.y2 / height,
                 )
-                text.text_representation = obj["text"]
+                text.text_representation = obj_text
                 texts.append(text)
         return texts
 
-    @timetrace("OCR Document Extraction")
+    @timetrace("OCRPageEx")
+    def extract_page(self, page: Optional[Union["PDFPage", "Image.Image"]]) -> list[Element]:
+        assert isinstance(page, Image.Image)
+        ocr_output = self.get_boxes_and_text(page)
+        width, height = page.size
+        texts: list[Element] = self.parse_ocr_output(ocr_output, width, height)
+        return texts
+
+    @timetrace("OCRDocEx")
     def extract_document(
         self, filename: Union[str, IOBase], hash_key: str, use_cache=False, **kwargs
-    ) -> List[List[Element]]:
-        cached_result = ocr_cache.get(hash_key) if use_cache else None
-        if cached_result:
+    ) -> list[list[Element]]:
+        if use_cache and (cached_result := ocr_cache.get(hash_key)):
             logger.info(f"Cache Hit for OCR. Cache hit-rate is {ocr_cache.get_hit_rate()}")
             return cached_result
-        else:
-            with tempfile.TemporaryDirectory() as tempdirname:  # type: ignore
-                filename = cast(str, filename)
-                images = kwargs.get("images")
-                generator = (image for image in images) if images else pdf_to_image_files(filename, Path(tempdirname))
-                pages = []
-                for path in generator:
-                    if isinstance(path, Image.Image):
-                        image = path
-                    else:
-                        image = Image.open(path).convert("RGB")
-                    ocr_output = self.get_boxes_and_text(image)
-                    width, height = image.size
-                    texts: List[Element] = []
-                    for obj in ocr_output:
-                        if obj["bbox"] and not obj["bbox"].is_empty() and obj["text"] and len(obj["text"]) > 0:
-                            text = Element()
-                            text.type = "text"
-                            text.bbox = BoundingBox(
-                                obj["bbox"].x1 / width,
-                                obj["bbox"].y1 / height,
-                                obj["bbox"].x2 / width,
-                                obj["bbox"].y2 / height,
-                            )
-                            text.text_representation = obj["text"]
-                            texts.append(text)
-
-                    pages.append(texts)
-                if use_cache:
-                    logger.info("Cache Miss for OCR. Storing the result to the cache.")
-                    ocr_cache.set(hash_key, pages)
-                return pages
+        with tempfile.TemporaryDirectory() as tempdirname:  # type: ignore
+            filename = cast(str, filename)
```

Comment:
It looks like pdf_to_image_files() doesn't accept an IOBase.  Should we assert that filename is an instance of str?

Reply:
Doesn't the cast deal with this already? We potentially get a wider range of acceptable entries this way.

Reply:
Changed.